### PR TITLE
[engine] Introduce Presto Reader for Paimon

### PR DESCRIPTION
**Pull request by s7monk**
@https://github.com/apache/incubator-paimon/issues/667
### Purpose

This PR aims to introduce presto reader for paimon.

### Tests

TestPrestoITCase

### API and Format 

No changes.

### Documentation

engines#presto

### CODE REVIEW

**Comment by JingsongLi** [comment]
Thanks @s7monk for the contribution, can we add more tests in `paimon-presto-common`, itcase is not enough.
We can add utcase for each classes.

File: paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoConnector.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import com.facebook.presto.spi.connector.EmptyConnectorCommitHandle;
+import com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorMetadata;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+
+import static com.facebook.presto.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static com.facebook.presto.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+/** Presto {@link Connector}. */
+public class PrestoConnector implements Connector {
```
Comment (JingsongLi) [inline paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoConnector.java:37]: classes in 0.273 can be in `paimon-presto-common`.

File: paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoPageSource.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+import static com.facebook.presto.common.type.Decimals.isShortDecimal;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.TimeType.TIME;
+import static com.facebook.presto.common.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static java.lang.String.format;
+
+/** Presto {@link ConnectorPageSource}. */
+public class PrestoPageSource implements ConnectorPageSource {
```
Comment (JingsongLi) [inline paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoPageSource.java:70]: Can we have more abstraction in this class? Like `PrestoPageSourceBase` to share some codes?

File: paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoConnector.java (hunk 1 above)
Comment (s7monk) [reply paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoConnector.java:37]: ok

File: paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoPageSource.java (hunk 2 above)
Comment (s7monk) [reply paimon-presto/paimon-presto-0.273/src/main/java/org/apache/paimon/presto/PrestoPageSource.java:70]: ok

**Comment by s7monk** [comment]
@JingsongLi  I have updated this PR, please help to review it again, thx

**Review by JingsongLi — COMMENTED** [review]
really thanks! I will take a look

File: docs/content/engines/presto.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Paimon currently supports Presto 0.236 and above.
+
+## Deploy Paimon Presto Connector
+
+Build bundled jar with the following command.
+
+|     Version      | Command                                                                                                                                                                                                            |
+|------------------|-------------------------------------------------------------------------|
+| [0.236,0.268)    | mvn clean install -Dmaven.test.skip=true -Ppresto-0.236                 |
```
Comment (JingsongLi) [inline docs/content/engines/presto.md:39]: Add download jars like spark 3?

File: docs/content/engines/presto.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+This documentation is a guide for using Paimon in Presto.
```
Comment (JingsongLi) [inline docs/content/engines/presto.md:27]: Also add presto in engines/overview too

File: paimon-presto/paimon-presto-common/src/test/resources/log4j2-test.properties [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# Set root logger level to OFF to not flood build logs
+# set manually to INFO for debugging purposes
+rootLogger.level = OFF
```
Comment (JingsongLi) [inline paimon-presto/paimon-presto-common/src/test/resources/log4j2-test.properties:21]: I can see logs still be printed.
Do we need a `airlift.log.Logger` configure file to remove logs?

File: paimon-presto/paimon-presto-0.273/pom.xml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>shade-flink</id>
```
Comment (JingsongLi) [inline paimon-presto/paimon-presto-0.273/pom.xml:123]: all shade-flink to shade-paimon

File: docs/content/engines/presto.md (hunk 3 above)
Comment (s7monk) [reply docs/content/engines/presto.md:39]: Thanks for your review，I will do this later.

File: docs/content/engines/presto.md (hunk 4 above)
Comment (s7monk) [reply docs/content/engines/presto.md:27]: ok

File: paimon-presto/paimon-presto-common/src/test/resources/log4j2-test.properties (hunk 5 above)
Comment (s7monk) [reply paimon-presto/paimon-presto-common/src/test/resources/log4j2-test.properties:21]: Solved a conflict, but currently I can't find a good way to remove logs

**Comment by JingsongLi** [comment]
I'v created a repository for presto. We can create PR in that repo.

**Comment by s7monk** [comment]
> 
ok

**Comment by s7monk** [comment]
> I'v created a repository for presto. We can create PR in that repo.

ok

**Comment by JingsongLi** [comment]
@s7monk https://github.com/apache/incubator-paimon-presto here

**Pull request closed without merging**